bin/ruby-build

```diff
@@ -1016,20 +1016,23 @@ has_broken_mac_openssl() {
   is_mac || return 1
   local openssl_version="$(/usr/bin/openssl version 2>/dev/null || true)"
   [[ $openssl_version = "OpenSSL 0.9.8"?* || $openssl_version = "LibreSSL"* ]] &&
-  [[ "$RUBY_CONFIGURE_OPTS" != *--with-openssl-dir=* ]] &&
```

Comment:
You've removed this line. This has served as a guard that Homebrew openssl never gets used if an explicit `--with-openssl-dir` was passed. You have removed this and placed the guard elsewhere. 

Do you have custom Ruby install formula by any chance?

Reply:
Consider the case that we have installed openssl@1.1 via Homebrew and want to build ruby with `--with-openssl-dir=` option.
Then this line is evaluated to truthy value, and thus `&&` invokes `use_homebrew_openssl`, resulting in setting `--with-openssl-dir="$ssldir"`.

Reply:
> Consider the case that we have installed openssl@1.1 via Homebrew and want to build ruby with `--with-openssl-dir=` option.
> Then this line is evaluated to truthy value, and thus `&&` invokes

If you had passed your own `--with-openssl-dir=` value, then the `[[ ... ]]` check evaluates to _false_, and nothing on the right side of `&&` is invoked.

Reply:
```sh
RUBY_CONFIGURE_OPTS="--with-openssl-dir=/foo/bar"

if [[ "$RUBY_CONFIGURE_OPTS" != *--with-openssl-dir=* ]]; then
  echo 'I will try to use Homebrew openssl!'
else
  echo 'I will NOT try to use Homebrew openssl.' # <-- this gets printed
fi
```

Reply:
Sorry, I completely misunderstood it...